marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/CallBatcherImpl.java

```diff
@@ -423,14 +424,18 @@ public Boolean call() {
 	public CallBatcherImpl<W, E> withdefaultArgs(CallArgs args) {
 		if(args == null)
 			throw new IllegalArgumentException("CallArgs cannot be null.");
+		
 		CallArgsImpl callArgsImpl = (CallArgsImpl) args;
+		
 		if(callArgsImpl == null || ! (callArgsImpl instanceof CallArgsImpl))
 			throw new IllegalArgumentException("Illegal Argument.");
```

Comment:
If callArgsImpl is null, should we return this for consistency with line 434? 

I believe the instanceof test must precede the cast.  Otherwise, the cast will fail before the instanceof test is executed.